manager/scheduler/constraint.go

```diff
@@ -27,11 +27,11 @@ func (f *ConstraintFilter) SetTask(t *api.Task) bool {
 func (f *ConstraintFilter) Check(n *NodeInfo) bool {
 	for _, constraint := range f.constraints {
 		switch constraint.Key {
-		case "node.id":
+		case "node.id", "engine.id":
```

Comment:
Let's make this `node.id` only. A node ID isn't an engine property in the same way these other `engine.` keys are.